=== modified file 'sql/backup/image_info.cc'
--- a/sql/backup/image_info.cc 2009-11-04 14:18:21 +0000
+++ b/sql/backup/image_info.cc 20092009-12-10 12:40:49 +0000
@@ -111,9 +111,9 @@ Image_info::~Image_info()
 */
 
 Image_info::Db*
-Image_info::add_db(const String &db_name, uint pos)
+Image_info::add_db(obs::Obj *db_obj, uint pos)
 {
-  Db *db= new (&mem_root) Db(db_name);
+  Db *db= new (&mem_root) Db(db_obj);
 
   if (!db)
     return NULL;
@@ -123,7 +123,18 @@ Image_info::add_db(const String &db_name
     m_dbs[pos]->~Db();
 
   if (m_dbs.insert(pos, db))
+  {
+    /*
+      We did not manage to successfully add db to the catalog. Thus we need 
+      to destroy it but first we set m_obj_ptr to NULL so that it is not 
+      deleted in the destructor. We assume ownership of the obs::Obj 
+      instance only if the object has been successfully added to the 
+      catalog.
+    */ 
+    db->m_obj_ptr= NULL;
+    db->~Db();
     return NULL;
+  }
 
   db->base.pos= pos;
 
@@ -144,9 +155,9 @@ Image_info::add_db(const String &db_name
 */
 
 Image_info::Ts*
-Image_info::add_ts(const String &ts_name, uint pos)
+Image_info::add_ts(obs::Obj *ts_obj, uint pos)
 {
-  Ts *ts= new (&mem_root) Ts(ts_name);
+  Ts *ts= new (&mem_root) Ts(ts_obj);
 
   if (!ts)
     return NULL;


=== modified file 'sql/backup/image_info.h'
--- a/sql/backup/image_info.h 2009-11-04 14:18:21 +0000
+++ b/sql/backup/image_info.h 2009-12-10 12:40:49 +0000
@@ -105,7 +105,8 @@ public: // Public interface.
 
   // Examine contents of the catalogue.
 
-  bool has_db(const String&) const;
+  Db* find_db(const String&) const;
+  Ts* find_ts(const String&) const;
 
   // Retrieve objects using their coordinates.
 
@@ -113,7 +114,6 @@ public: // Public interface.
   Ts*    get_ts(uint pos) const;
   Dbobj* get_db_object(uint db_num, ulong pos) const;
   Table* get_table(ushort snap_num, ulong pos) const;
-  Db*    find_db(const String &db_name);


=== modified file 'mysql-test/suite/backup_extra/vptest/run_vp_test.pl'
--- a/mysql-test/suite/backup_extra/vptest/run_vp_test.pl   2009-11-19 15:50:00 +0000
+++ b/mysql-test/suite/backup_extra/vptest/run_vp_test.pl   2009-12-07 08:34:04 +0000
@@ -567,7 +567,7 @@ sub execute_sql {
     my($port,$sql) =@_;
 
     my $do_sql = "END_OF_SQL\n".$sql."\nEND_OF_SQL\n";
-    $mysql_args="--user=root --protocol=tcp --port=".$port;
+    $mysql_args="--skip-column-names --user=root --protocol=tcp --port=".$port;
     my $command = $client_dir."/mysql ".$mysql_args.
         " << ".$do_sql;


=== modified file 'mysql-test/suite/backup/t/backup_bml.test'
--- a/mysql-test/suite/backup/t/backup_bml.test 2009-06-30 07:51:04 +0000
+++ b/mysql-test/suite/backup/t/backup_bml.test 2009-11-26 09:52:48 +0000
@@ -36,6 +36,11 @@
 ##       main, a copy of this test with a modified part 3 has been
 ##       created as backup_bml_not_falcon.test. Once Falcon is back in
 ##       main, that test file should be removed.
+## NOTE: The result file of backup_bml_not_falcon has since changed.
+##       Without Falcon, the result file for this test can't be
+##       recorded. When reenabling this test, please check the diff
+##       between the result file of this test and
+##       backup_bml_not_falcon.result
